A Brand New wasabiwallet.io ⭐️

mayankchhabra opened the pull request:
This PR redesigns the current website of Wasabi Wallet with a fresh new interface, experience, functionality, and aesthetics. 🎉

All the theme-wide changes, such as the color scheme, typography, spacings, links, buttons, etc are made via a custom Bootstrap 4.3 build using SASS variables, included in the `_variables.scss` file in `WalletWasabi.Backend/wwwroot/css/` directory. 

This has two main benefits in particular: 
1. It allows us to carry the same Bootstrap theme across all WW websites, such as the upcoming docs with ease
2. It removes the need for unnecessary code overwrites in `style.css`

Custom CSS in `style.css` has been written in the only cases where there is no default Bootstrap class or structure available to produce the desired outcome, hence the new `style.css` file is marginally larger, 2KB to be precise, than the current one when compressed.

A test server is live with the up-to-date website code at [weareblock.space/wasabi](https://weareblock.space/wasabi) until this PR is merged. Please test it across your devices with different browsers and report any issues, feedback or suggestions. 

To-do before merge: 
- [ ] ~~Implement dynamic loading of the latest CoinJoin transactions~~
CJ transactions are now hardcoded (not live), providing the "proofing" and educating user with live examples, at the same time removing the `<iframe>` <-> server hack. 

- [ ] ~~Implement dynamic loading of the latest release version in the downloads section~~
Latest release version will be manually updated at every release, as the download links, signature and guides were updated manually anyway, removing the last bridge between the site and the server. 

- [x] Update textual content

- [x] Fix any bugs

MaxHillebrand commented:
Oh this looks so good!! Thank you very much for this @mayankchhabra :heart:!!

A couple things:

- [ ] at the bottom the `Press Kit` is split awkwardly in two lines when the window is at about 60% of screen width
- [ ] It would be great to link to the new `docs.wasabiwallet.io` website in the top nav bar for `dojo` and `faq` or just `docs` in general, but the link is only ready when @nopara73 does the CNAME entry.
- [ ] the bottom FAQ is a good selection of questions - and below it should link to the new docs website. 
- [ ] the coin join link should probably go to Blockstream.info, or what other block explorer do you suggest?

MaxHillebrand commented:
May I suggest to make the necessary ToDos to release this new website and the [new documentation](https://github.com/zkSNACKs/WasabiDoc/projects/1) a short term priority? It does not have to be perfect, we can always push fixes and improvements.

I would really really like if we can push this on August 1st to celebrate #UASF.

nopara73 commented:
> I would really really like if we can push this on August 1st to celebrate #UASF.

@MaxHillebrand Wasabi's first beta release was also on August 1:)

nopara73 commented:
@mayankchhabra This is great! Are you looking for review or not yet?  

As I see it, one of your main issue is/will be how to show the recent coinjoins properly. Maybe we could remove that part altogether, because at this point nobody will be convinced that Wasabi works based on that data, which was the reason why it got there in the first place, but there's enough press for this to not be a questionable anymore.

mayankchhabra commented:
Thanks, Max and Adam! 

Alright then, seems like August 1 really is the day with the universal consensus around it, haha!

Working on your suggestions, Max. 

Adam, yes let's get this reviewed. 

> because at this point nobody will be convinced that Wasabi works based on that data, which was the reason why it got there in the first place, but there's enough press for this to not be a questionable anymore 

ACK that there is no proofing required anymore as there is enough press-coverage and social-proofing via twitter, etc. Also, with the current back-end code, it's not viable to load up an `<iframe>` with duplicate CSS files to just show the transactions. 

However, it's nice to see it live in action, as that gives the sense of how it looks in real life, plus educates users with live, real-life examples. What we can do is, for now, hard code some of the CJ txs done by Wasabi, including the famous 100 participant tx, and then work on the backend endpoint later on for fetching live txs. What do you think? 

Also, dynamic release number for downloads are better done as manual updates with every release, instead of loading an `<iframe>` since the download, signature, and installation guides are manually update anyway with every release for all binaries. 

Lastly, let's get the textual content for the CJ paragraph and FAQs in place so we can get this up!

mayankchhabra commented:
Ready for review and merge @nopara73 @MaxHillebrand 🚀

nopara73 commented:
@lontivero @molnard @jmacato could one of you also review this?

MaxHillebrand commented:
Ok, we finally have docs.wasabiwallet.io live and on mainnet!!
Now we can fix the links in this main website: 

- upper menu "docs" - https://docs.wasabiwallet.io
- upper menu "faq" - https://docs.wasabiwallet.io/FAQ/
- upper menu "dojo" - https://docs.wasabiwallet.io/building-wasabi/Dojo.html

It might be nice to add a link to the installation instructions of:
- "windows" - https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html#windows
- "debian & ubuntu" - https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html#debian-and-ubuntu
- "other linux" - https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html#other-linux
- "mac" - https://docs.wasabiwallet.io/using-wasabi/InstallPackage.html#osx

MaxHillebrand commented:
How about a dark mode for the site? I think this would make it look :100:+ better, and it can be opt-in.
@dennisreimann has implemented in [docs.wasabiwallet.io](https://docs.wasabiwallet.io) this adaptive dark mode that is activated when the user has system preferences for not getting his eyes blinded by a white screen.

https://github.com/zkSNACKs/WasabiDoc/pull/53#issuecomment-517945361

MaxHillebrand commented:
The four panels are great, but I feel the urge to click them - should they be hyperlinked? I think this is a good place to link to the docs with the easy introduction to each of these problems, and how Wasabi solves them. 

- In-Built CoinJoins > https://docs.wasabiwallet.io/why-wasabi/BitcoinPrivacy.html#transaction-graph
- Tor Integration > https://docs.wasabiwallet.io/why-wasabi/BitcoinPrivacy.html#network-snooping
- Coin-Control Features > https://docs.wasabiwallet.io/why-wasabi/BitcoinPrivacy.html#inputs-and-outputs
- Free & Open Source > https://github.com/zksnacks/walletwasabi

MaxHillebrand commented:
> Wasabi automates the process of coin-shuffling using Chaumian CoinJoins over the Tor network, allowing you to mix your coins with other Wasabi users in a purely trustless manner **without relying on any third party**.

But the CoinJoin DOES rely on a third party to coordinate the communication and transaction building. Without the coordinator, the coin join would not be successful.

However, the coordinator **cannot steal**, meaning the user sends the money atomically to himself, and not to a trusted third party; and **cannot spy**, meaning due to Schnorr blind signatures, tor and block filter network level privacy, the user never reveals input-to-output link to anyone.

mayankchhabra commented:
@nopara73 awaiting your thoughts on "an unfairly private wallet for Bitcoin" being the headline vs "Reclaim your privacy now". As mentioned in Slack, I think the former is a better-suited text for the landing page, because as soon as the user lands, he should understand, without the need of reading a paragraph of text, what the product is about in a short and sweet manner. 

"Reclaim your privacy" has been WW's go-to slogan since the beginning and should still stick as our tagline and CTA. Hence when you share the link anywhere, a nice card pops up with the image/text complimenting each other. Image attached:

![image](https://user-images.githubusercontent.com/10330103/62423005-31791000-b6ee-11e9-9b4b-e7b00043d7d6.png)


Same for the title, instead of "unfairly private". Having "Bitcoin" in the name is important for both SEO and understanding purposes, as Wasabi is exclusive to BTC only. 

***

@MaxHillebrand thanks for your feedback and suggestions! 

Regarding 
> upper menu "faq" - https://docs.wasabiwallet.io/FAQ/

Since the website has a FAQ section, replacing the Navbar's "faq" with the link to the docs could be confusing. I suggest we change "Can't find your question?..." at the bottom of the website's FAQ to "Can't find your question? Read our extended FAQ in docs >"

What do you think?

> It might be nice to add a link to the installation instructions

Absolutely! We do have installation instructions on the website. Under every download button, there's a help button but that leads to the old instructions on the repo, not the new docs. So I will update the links. Also, I think calling the links "Guide" is better suited than "Help".

> How about a dark mode for the site?

Oh yes! Let's figure that out especially when we can do a preferential dark-mode like the docs now!

> The four panels are great, but I feel the urge to click them - should they be hyperlinked?

Why not is the question! Let's seize every moment (or should I say every urge) to educate! 🔥 

> But the CoinJoin DOES rely on a third party

Spot on! Let's do away with "without relying on any third party".

mayankchhabra commented:
New version pushed to the [test server](https://weareblock.space/wasabi) with the following changes:
- [x] Fixed hero section's text cut on smaller heights
- [x] The 4 feature cards are now links
- [x] Fixed broken links
- [x] Added installation guide links
- [x] Added extended FAQs link under the FAQ section
- [x] Added a learn more link in CoinJoins section

dennisreimann commented:
Yeah, I second that – really gorgeous! Superb look 👍🏻🥰

nopara73 commented:
# Behavior Review

It's perfect. One note: (mainly to @MaxHillebrand) It has a lot of links and they will be broken. Unfortunately it ruins the maintainability.

I'll proceed wit the core review now.

nopara73 commented:
# Code Review

# Existing Codebase changes

- [x] No changes were made other than removing old files.

# Visual Studio Complains

- [ ] There are about Visual Studio complains. I would like all of them to be fixed, but when it it's not possible then discuss them and get to the end of the code smell.

![image](https://user-images.githubusercontent.com/9156103/62459468-b6f8d080-b77f-11e9-8253-1d01f81bd6b0.png)

How shall we proceed with this? @mayankchhabra Do you have access to Visual Studio or shall one of us gather them out here?

# Other notes:

I will deploy this to the testnet now and see if the Tor version works, too.

mayankchhabra commented:
Thanks, @MaxHillebrand, @nopara73, and @dennisreimann for your feedback! 🙌

Pushed an update:
- [x] Fixed docs link in the navbar
- [x] Rectified important w3c validation errors in index.html

@nopara73 VS 2017 has lousy IntelliSense when it comes to HTML/CSS validation. As you can see, most of the validation errors are in-fact from bootstrap.css. 

The general rule of thumb is that if the browsers render everything as you expect, just go with it. Browsers are updated with new CSS properties quite often with their every release, while VS waits for an official CSS release that might take months if not years. So I'd say we can comfortably ignore the validation errors and go forward! 🚀

A couple of things that I'd like to suggest before we merge:

- [ ] Having the word "Bitcoin" in all title tags. Right now it is `Wasabi Wallet - Unfairly Private`, we can change it to `Wallet Wallet - An unfairly private wallet for Bitcoin`. Instantly gets the idea across, plus helps in SEO. 

- [ ] Changing the heading of the website from "reclaim your privacy now" back to "An unfairly private wallet for Bitcoin". Same reason, the former is a strong CTA slogan, the latter is a product description. Under "An unfairly private wallet for Bitcoin" we can write "reclaim your privacy now" as the tagline to push the narrative forward with the description and CTA duo complementing each other. 

- [ ] The heading of the website on mobile changes from "reclaim your privacy now" to "unfairly private". I'm not sure if that was intentional @nopara73  or you accidentally changed it while changing the title tags...

mayankchhabra commented:
> Read our extended FAQs in **the documentation** 

We can change that, but the text becomes really long as a single link itself... That's why it was trimmed to keep it short. Perhaps, we can do `<no link>Couldn't find your question?</no link> <link>Read extended FAQs in the documentation</link>`

Regarding creating a separate repo for the hosting, I do agree that the website would be better suited to have its own repo, more so because we do not have a client-server link anymore that utilizes the `Wasabi.Backend` APIs. However, I think instead of merging the website with the docs repo, it's better to create a `WasabiWebsite` repo for more organized, productive and efficient management.

nopara73 commented:
>  Having the word "Bitcoin" in all title tags. Right now it is Wasabi Wallet - Unfairly Private, we can change it to Wallet Wallet - An unfairly private wallet for Bitcoin. Instantly gets the idea across, plus helps in SEO.

Concept ACK, but implementation NACK. The problem with the "An unfairly private wallet for Bitcoin" is that "unfairly private" is dubious enough if makes grammatical sense. It's a reference to Jihan calling segwit "unfairly cheap" so we should use it as is or just change it altogether.

> Changing the heading of the website from "reclaim your privacy now" back to "An unfairly private wallet for Bitcoin". Same reason, the former is a strong CTA slogan, the latter is a product description. Under "An unfairly private wallet for Bitcoin" we can write "reclaim your privacy now" as the tagline to push the narrative forward with the description and CTA duo complementing each other.

Same comment.

> The heading of the website on mobile changes from "reclaim your privacy now" to "unfairly private". I'm not sure if that was intentional @nopara73 or you accidentally changed it while changing the title tags...

Yes, it looked better when I restored the original quote.


# Summary

I don't have a good idea how to keep Bitcoin in the texts. "unfairly private" should be kept intact, only change it if there's a better text to put there.


@MaxHillebrand 

> And a general question, this PR would merge the new website in the main zkSnacks/WalletWasabi repository, yet this clutters the code, and it is a hindrance for fast and reckless iteration of the website in the future. Should we instead host this in the new zkSnacks/WasabiDoc repository?

We can't. The Tor website must be ran from the server, because that's the same url the clients are talking to.

nopara73 commented:
I'd love one more review on this. Anyone could do it just check the testnet website so no need to set it up: https://wasabiwallet.co/

mayankchhabra commented:
Great! @nopara73 can you please push the latest code on the testnet server? The formatting for the docs link underneath the FAQ has been tweaked a bit.

benthecarman commented on WalletWasabi.Backend/WebsiteTorifier.cs:
```diff
@@ -0,0 +1,32 @@
+
+			var content = File.ReadAllText(path);
+
+			content = content.Replace("http://wasabiukrxmkdgve5kynjztuovbg43uxcbcxn6y2okcrsg7gb6jdmbad.onion", "https://wasabiwallet.io", StringComparison.Ordinal);
```
The onion link might be able to be moved to a variable, it is also linked in the About view page.

benthecarman commented:
The dojo link in top toolbar links to https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Community/Dojo.md, it should link to https://docs.wasabiwallet.io/building-wasabi/Dojo.html

benthecarman commented:
The build from source link points to the repo, it would be better if it pointed to https://github.com/zkSNACKs/WalletWasabi#build-from-source-code

benthecarman commented:
The free and open source button sends to the repo, maybe consider changing it to https://docs.wasabiwallet.io/why-wasabi/10Commandments.html#_2-verify-the-integrity-of-your-software

mayankchhabra commented:
Good feedback @benthecarman, thank you! The links have been changed. Waiting for @nopara73 to redeploy on testnet for the final testing. 🚀

benthecarman commented:
The "Further reading: [Reddit] Combining Mixed Coins" and "Further reading: Chaumian Coinjoin" just go to wasabiwallet.io

mayankchhabra commented:
Good find @benthecarman 🔍
Fixed the links!

nopara73 commented:
Thank you guys, great work, this may have been the first large PR I didn't feel the need to micromanage :)

I deployed it to the testnet. We'll deploy it to the mainnet with the next release.

https://wasabiwallet.co/

yahiheb commented:
Great work guys.
